### PR TITLE
Nicogbg/ms kubernetes

### PR DESCRIPTION
**Pull request by nicogbg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x ] The toolchain/worker package manifests are up-to-date
- [x ] Any updated packages successfully build (or no packages were changed)
- [x ] All package sources are available
- [ x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [ x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ x] All source files have up-to-date hashes in the `*.signatures.json` files
- [ x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x ] Documentation has been updated to match any changes to the build system
- [x ] Ready to merge

---

###### Summary 

remove kubernetes 1.16 and create kubernetes to 1.17, 1.18 and 1.19

###### Change Log  

spec files under SPECS/kubernetes

###### Does this affect the toolchain?  

NO

### CODE REVIEW

File: SPECS/kubernetes/kubernetes-1.17.spec [hunk 1]
```diff
@@ -0,0 +1,207 @@
+%global debug_package %{nil}
+%ifarch x86_64
+%define archname amd64
+%endif
+%ifarch aarch64
+%define archname arm64
+%endif
+%define kubeproxy_n_1 1.16.13
```
Comment (jiria) [inline SPECS/kubernetes/kubernetes-1.17.spec:8]: Curious, what do we need the kubeproxy_n_1 and coredns_n_1 for?

File: SPECS/kubernetes/kubernetes-1.17.spec [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+#       by doing nothing in %build and %check sections
+%build
+# expand kubernetes source tarball (which is included source0 tarball)
+echo "+++ extract sources from tarball"
+mkdir -p %{_builddir}/%{name}/src
+cd %{_builddir}/%{name}/src
+tar -xof %{_builddir}/%{name}/kubernetes-src.tar.gz
+
+# build and update kubernetes components that are provided as binary
+# (other/unused kubernetes componenents will not be built)
```
Comment (jiria) [inline SPECS/kubernetes/kubernetes-1.17.spec:72]: nit: components

File: SPECS/kubernetes/kubernetes-1.17.spec [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+components_to_build=$(ls -1 %{_builddir}/%{name}/node/bin)
```
Comment (jiria) [inline SPECS/kubernetes/kubernetes-1.17.spec:73]: what does components_to_build evaluates to? We should not be building apiserver and such here imho.

File: SPECS/kubernetes/kubernetes-1.17.spec [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+if [ $1 -eq 0 ]; then
+    # Package deletion
+    userdel kube
+    groupdel kube
+    systemctl daemon-reload
+fi
+
+%files
+%defattr(-,root,root)
+%license LICENSES
```
Comment (jiria) [inline SPECS/kubernetes/kubernetes-1.17.spec:158]: Where is LICENCES coming from?

File: SPECS/kubernetes/kubernetes-1.17.spec (hunk 1 above)
Comment (nicogbg) [reply SPECS/kubernetes/kubernetes-1.17.spec:8]: I just left what was in the 1.17 spec file in msk8s. I actually don't know if it is needed (does not seem used anywhere in the .spec). I'm OK to remove it.

File: SPECS/kubernetes/kubernetes-1.17.spec (hunk 3 above)
Comment (nicogbg) [reply SPECS/kubernetes/kubernetes-1.17.spec:73]: this variable contain the list of the kubernetes binaries found in node/bin in kubernetes-node-linux-amd64.tar.gz tarball (source0).
The goal is to only rebuild the binaries that the upstream provide in the tarball

File: SPECS/kubernetes/kubernetes-1.17.spec (hunk 4 above)
Comment (nicogbg) [reply SPECS/kubernetes/kubernetes-1.17.spec:158]: the LICENCES file come from kubernetes-node-linux-amd64.tar.gz tarball. By default %license macro use file from the root folder where source0 tarball has been expended.
and yes it took me a while to realize that ;)

File: SPECS/kubernetes/kubernetes-1.17.spec (hunk 1 above)
Comment (nicogbg) [reply SPECS/kubernetes/kubernetes-1.17.spec]: discussed off line with George and these are left over => I'll remove that

File: SPECS/kubernetes/kubernetes-1.17.spec (hunk 2 above)
Comment (nicogbg) [reply SPECS/kubernetes/kubernetes-1.17.spec]: fixed

File: SPECS/kubernetes/kubernetes-1.17.spec [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+Summary:        Microsoft Kubernetes
+Name:           kubernetes
+Version:        1.17.11
+Release:        3%{?dist}
+License:        ASL 2.0
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          Microsoft Kubernetes
+URL:            https://mcr.microsoft.com/oss
+#               Download URL https://kubernetesartifacts.azureedge.net/kubernetes/v1.17.11-hotfix.20200901/binaries/kubernetes-node-linux-amd64.tar.gz
```
Comment (schmittjoseph) [inline SPECS/kubernetes/kubernetes-1.17.spec:17]: Our convention with instances where the SourceX url doesn't perfectly match the source name is to put a `"#SourceX: url"` above it. In this case, it'd be replacing the "Downloading URL" with:

`#Source0: https://kubernetesartifacts.azureedge.net/kubernetes/v1.17.11-hotfix.20200901/binaries/kubernetes-node-linux-amd64.tar.gz`

Applies to the other SPECs changed in this PR as well.

**Review by jiria — APPROVED** [review]

File: SPECS/kubernetes/kubernetes-1.17.spec (hunk 5 above)
Comment (nicogbg) [reply SPECS/kubernetes/kubernetes-1.17.spec:19]: I didn't know that, I'll fix it

**Review by schmittjoseph — APPROVED** [review]

**Pull request merged**